pkg/node/node.go

```diff
@@ -263,6 +263,11 @@ func NewBee(interrupt chan struct{}, addr string, publicKey *ecdsa.PublicKey, si
 	var batchStore postage.Storer = new(postage.NoOpBatchStore)
 	var unreserveFn func([]byte, uint8) (uint64, error)
 
+	post, err := postage.NewService(stateStore, batchStore, chainID)
```

Comment:
This is wrong. This will initialize the postage service with NoOpBatchStore.

What you should do is call batchStore.SetBatchExpiryHandler after the postage service is created instead of moving this here.